### PR TITLE
Support UNTESTED test state

### PR DESCRIPTION
**Pull request by odubaj**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -65,9 +65,12 @@ dependencies {
         compile 'com.epam.reportportal:commons-fonts'
         compile 'com.epam.reportportal:plugin-api'
     } else {
-        compile 'com.github.reportportal:commons-dao:f786006'
+        compile 'com.github.odubaj:commons-dao:bbcf648'
```
Comment (Yumfriez) [inline build.gradle:68]: will be replaced on our dependency after approval

File: build.gradle [hunk 2]
```diff
@@ -65,9 +65,12 @@ dependencies {
...
+        //compile fileTree(dir: '/home/odubaj/DIPLOMKA/commons-dao/build/libs', include: ['*.jar'])
         compile 'com.github.reportportal:commons-rules:331c402'
-        compile 'com.github.reportportal:commons-model:2662a59b'
+        //compile fileTree(dir: '/home/odubaj/DIPLOMKA/commons-rules/build/libs', include: ['*.jar'])
+        compile 'com.github.odubaj:commons-model:5b47278'
```
Comment (Yumfriez) [inline build.gradle:72]: will be replaced on our dependency after approval

File: build.gradle [hunk 3]
```diff
@@ -65,9 +65,12 @@ dependencies {
...
+        //compile fileTree(dir: '/home/odubaj/DIPLOMKA/commons-model/build/libs', include: ['*.jar'])
```
Comment (Yumfriez) [inline build.gradle:73]: remove comments please

File: build.gradle [hunk 4]
```diff
@@ -65,9 +65,12 @@ dependencies {
         compile 'com.epam.reportportal:commons-fonts'
         compile 'com.epam.reportportal:plugin-api'
     } else {
-        compile 'com.github.reportportal:commons-dao:f786006'
+        compile 'com.github.odubaj:commons-dao:bbcf648'
+        //compile fileTree(dir: '/home/odubaj/DIPLOMKA/commons-dao/build/libs', include: ['*.jar'])
         compile 'com.github.reportportal:commons-rules:331c402'
-        compile 'com.github.reportportal:commons-model:2662a59b'
+        //compile fileTree(dir: '/home/odubaj/DIPLOMKA/commons-rules/build/libs', include: ['*.jar'])
```
Comment (Yumfriez) [inline build.gradle:71]: remove comments please

File: build.gradle [hunk 5]
```diff
@@ -65,9 +65,12 @@ dependencies {
         compile 'com.epam.reportportal:commons-fonts'
         compile 'com.epam.reportportal:plugin-api'
     } else {
-        compile 'com.github.reportportal:commons-dao:f786006'
+        compile 'com.github.odubaj:commons-dao:bbcf648'
+        //compile fileTree(dir: '/home/odubaj/DIPLOMKA/commons-dao/build/libs', include: ['*.jar'])
```
Comment (Yumfriez) [inline build.gradle:69]: remove comments please

File: build.gradle [hunk 6]
```diff
@@ -76,6 +79,7 @@ dependencies {
     compile 'org.springframework.boot:spring-boot-starter-aop'
     compile 'org.springframework.boot:spring-boot-starter-web'
     compile 'org.springframework.boot:spring-boot-starter-quartz'
+    compile 'org.springframework.boot:spring-boot-starter-jdbc'
```
Comment (Yumfriez) [inline build.gradle:82]: this dependency is not required here

File: project-properties.gradle [hunk 7]
```diff
@@ -2,7 +2,7 @@ sourceCompatibility = JavaVersion.VERSION_11
 targetCompatibility = JavaVersion.VERSION_11
 
 def commonScriptsUrl = 'https://raw.githubusercontent.com/reportportal/gradle-scripts/'
-def migrationsScriptsUrl = 'https://raw.githubusercontent.com/reportportal/migrations/'
+def migrationsScriptsUrl = 'https://raw.githubusercontent.com/odubaj/migrations/'
```
Comment (Yumfriez) [inline project-properties.gradle:5]: will be replaced on our dependency after approval

File: src/main/java/com/epam/ta/reportportal/core/item/impl/status/ToFailedStatusChangingStrategy.java [hunk 8]
```diff
@@ -74,9 +75,10 @@ protected void updateStatus(Project project, Launch launch, TestItem testItem, S
 
 		testItem.getItemResults().setStatus(providedStatus);
 		if (Objects.isNull(testItem.getRetryOf())) {
-			if (testItem.getItemResults().getIssue() == null && testItem.isHasStats()) {
-				addToInvestigateIssue(testItem, project.getId());
-			}
+			ofNullable(testItem.getItemResults().getIssue()).ifPresent(issue -> {
```
Comment (Yumfriez) [inline src/main/java/com/epam/ta/reportportal/core/item/impl/status/ToFailedStatusChangingStrategy.java:78]: this logic should not be changed:
we may have item with status "SKIPPED" and issue with types: ab, pb, etc. This issue type should not be removed if it exists

**Review by Yumfriez — CHANGES_REQUESTED** [review]
also update your branch with latest changes please

**Comment by odubaj** [comment]
Requested changes resolved

File: src/main/java/com/epam/ta/reportportal/core/item/impl/status/ToFailedStatusChangingStrategy.java (hunk 8 above)
Comment (odubaj) [reply src/main/java/com/epam/ta/reportportal/core/item/impl/status/ToFailedStatusChangingStrategy.java]: Ok, thanks, also fixed this in ToUntestedChangingStrategy.java

**Comment by odubaj** [comment]
Hello, any updates here ?

**Pull request merged**